src/averageAges.js

```diff
@@ -15,11 +15,16 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const malesBornInCentury = people.filter(person => person.sex === 'm'
+  && Math.ceil(person.died / 100) === century);
+
+  const allMales = people.filter(person => person.sex === 'm');
+
+  const males = century ? malesBornInCentury : allMales;
```

Comment:
first you can find all men and then if century is passed you can use the same array to filter males born in specific century
now you filter people array each time